classes/form/CustomerAddressForm.php

```diff
@@ -92,23 +92,27 @@ public function loadAddressById($id_address)
 
     public function fillWith(array $params = [])
     {
-        // This form is very tricky: fields may change depending on which
-        // country is being submitted!
-        // So we first update the format if a new id_country was set.
-        // If detect country from browser language is selected, default country will be overridden
-        if (Tools::isCountryFromBrowserAvailable()) {
+        // This form is tricky: fields may change depending on which country is being selected!
+        // Country preselection priority order :
+        // 1) Update the format if a new id_country was set.
+        // 2) Detect country from browser language settings and matches BO enabled countries
+        // 3) Default country set in BO
+        if (isset($params['id_country']) && $params['id_country'] != $this->formatter->getCountry()->id) {
+            $this->formatter->setCountry(
+                new Country($params['id_country'], $this->language->id)
+            );
+        } elseif (Tools::isCountryFromBrowserAvailable() && Country::getByIso(Tools::getCountryIsoCodeFromHeader(), true)) {
             $languageAvailable = Tools::getCountryIsoCodeFromHeader();
```

Comment:
could we use this occasion to rename this variable?
I would say `$countryIsoCode` would reflect the value much better.

Reply:
Agreed